com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/StepsUtil.java

```diff
@@ -406,6 +407,36 @@ public static void setAppsToDeploy(DelegateExecution context, List<CloudApplicat
             .collect(Collectors.toList());
         context.setVariable(Constants.VAR_APPS_TO_DEPLOY, cloudApplicationsAsStrings);
     }
+    
+    @SuppressWarnings("unchecked")
+    public static List<CloudApplicationExtended> getIteratedAppsInParallel(DelegateExecution context) {
+        Object appsDeployedInParallel = context.getVariable(Constants.VAR_ITERATED_APPS_IN_PARALLEL);
+        if (appsDeployedInParallel == null) {
+            return new ArrayList<>();

```

Comment:
This line makes the method know more than it should. It "knows" that people would add things to this collection. Replace this line with "return Collections.emptyList();" and use ListUtils.union in the step where you're adding more apps to the collection.